articles/overview/what-is-qsharp.md

```diff
@@ -0,0 +1,41 @@
+---
+title: What is Q#?
+description: Learn about the Q#, a programming language created by Microsoft to develop applications for quantum computers
+author: natke
+ms.author:  nakersha
+ms.date: 10/16/2019
+ms.topic: article
+uid: microsoft.quantum.overview.qsharp
+---
+
+# What is Q#?
+
+Q# is a programming language created by Microsoft to develop applications for quantum computers.
+
+The Q# programming language takes care of all the quantum computer's internal logic and provides you with an intuitive set of types, operations and logic expressions to develop algorithms without having to worry about the internal logic of the quantum computer.
+
+## We want to code algorithms, not to draw circuits
+
+In [What is quantum computing?](xref:microsoft.quantum.overview.what) we saw that we can write quantum computations as quantum circuits like we do in classical computing. The circuit model has been very useful for many years in quantum computing research.
+
+However, when we write classical programs for normal computers, we don’t think in terms of circuits but in terms of logic.
+
+Here in Microsoft, we believe in empowering developers to go beyond quantum circuits by offering the tools to develop quantum algorithms. Q# allows you to use the knowledge of other high level programming languages and apply them to developing applications on quantum computers.
+
+## How does Q# work?
+
+Q# is a programming language designed to be great a writing quantum programs, by providing features that are special to quantum computing. Q# provides quantum programmers a framework that allows them to focus on the algorithms without having to care about technical details like gate sequence optimization or the physical implementation of a quantum computer.
+
+For example, one of the fundamental building blocks of Q# is the `Qubit` type, which cannot be copied or directly accessed, just like a real qubit. Instead, we can measure it and store the outcome of the measurement in a `Result` variable, a Q# type that can take two possible values: `Zero` and `One`. Constructs like this one guarantee that algorithms always respect the laws of quantum mechanics and can run correctly on quantum computers or simulators.
+
+Q# also includes classical logic features like conditionals or loops with some subtleties to make sure that all the quantum rules are being respected. For example, quantum operations need to be reversible. This enforces some constraints on the way loops are executed.
+
+Q# programs are often paired with a host program written in C# or Python, which can provide convenient organization of classical and quantum code. In addition to supporting .NET languages such as C# and Python, the QDK provides Jupyter Notebook support with the IQ# Jupyter kernel.
```

Comment:
We should explain this a bit more in detail. There are new comments about this in the word document. We can talk about it in today's meeting.